### DISCUSSION TITLE
x86 vs x64 in .NET

### DISCUSSION THREAD

**Discussion by giscus** [linkdotnet/Blog.Discussions#21] [General]
# x86 vs x64 in .NET

In this article I will show you what is the difference between x86 (32-bit) vs x64 (64-bit) in the .NET World.
What is the impact if you choose on or another.

https://steven-giesel.com/blogPost/392d8179-c02c-4d7d-897d-e6a055c970b9

**Comment by BryanEllis** [comment]
Good article. Having worked in PCs since 16-bit and dealing with thunking over the years it is good to know why we still need the specific modes instead of always using AnyCPU.
Sadly, the numerous incorrect words make it more difficult to read. It seems like the article was dictated and the speech-to-text translation put in perfectly spelled words that sound similar to the ones that were spoken but are not.
Example: Under Garbage Collector it has "blogs of memory" instead of "blocks of memory"
Also, "neglactable" is not a word (even the in-browser Microsoft Editor keeps forcing it to "neglectable" as I try to comment).

**Reply by linkdotnet** [reply]
Thanks for the repot @BryanEllis.
Sometimes I should read my articles once more. Anyway  I tried to correct as many mistakes as I found.

**Comment by darthwalsh** [comment]
This was interesting to read! One question about:

> if you compile your .NET assemblies natively for x64 you will also notice that your assemblies are bigger. 

Is this saying if you have a purely managed DLL, changing the bitness changes the file size? Or just the JIT size in memory?

**Reply by linkdotnet** [reply]
I will address this better in the blog post. To your question:
Purely managed dll's aka libraries will **not** see an increase in filesize, but will see increase of RAM use (JIT translated, for example function pointer to 8 byte addresses instead of 4 byte addresses).

I will correct the quoted part from you to:
> if you compile your .NET **executables** natively ...

Executables are getting the direct hit.

**Reply by linkdotnet** [reply]
I updated the passage in the text. Thanks again for the clarifying question @darthwalsh
